### PR TITLE
[libgraphicsmagic] Add more optional deps to expand functionality

### PR DESCRIPTION
**Pull request by Moelf**
adding more optional deps to expand functionality

### CODE REVIEW

File: L/libgraphicsmagic/build_tarballs.jl [hunk 1]
```diff
@@ -34,8 +34,19 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
+    Dependency(PackageSpec(name="libwebp_jll", uuid="c5f90fcd-3b7e-5836-afba-fc50a0988cb2"))
+    Dependency(PackageSpec(name="libpng_jll", uuid="b53b4c65-9356-5827-b1ea-8c7a1a84506f"))
```
Comment (giordano) [inline L/libgraphicsmagic/build_tarballs.jl:39]: Hold on, today I discovered an incompatibility in libpng, you'll have to specify the version

File: L/libgraphicsmagic/build_tarballs.jl [hunk 2]
```diff
@@ -34,8 +34,19 @@ products = [
...
+    Dependency(PackageSpec(name="FreeType2_jll", uuid="d7e528f0-a631-5988-bf34-fe36492bcfd7"))
+    Dependency(PackageSpec(name="Ghostscript_jll", uuid="61579ee1-b43e-5ca0-a5da-69d92c66a64b"))
+    Dependency(PackageSpec(name="Xorg_libXext_jll", uuid="1082639a-0dae-5f34-9b06-72781eeb8cb3"))
+    Dependency(PackageSpec(name="Xorg_libSM_jll", uuid="c834827a-8449-5923-a945-d239c165b7dd"))
+    Dependency(PackageSpec(name="Libtiff_jll", uuid="89763e89-9b03-5906-acba-b20f662cd828"))
+    Dependency(PackageSpec(name="LittleCMS_jll", uuid="d3a379c0-f9a3-5b72-a4c0-6bf4d2e8af0f"))
+    Dependency(PackageSpec(name="Bzip2_jll", uuid="6e34b625-4abd-537c-b88f-471c36dfa7a0"))
```
Comment (giordano) [inline L/libgraphicsmagic/build_tarballs.jl:46]: https://github.com/JuliaPackaging/Yggdrasil/issues/3024

File: L/libgraphicsmagic/build_tarballs.jl (hunk 1 above)
Comment (Moelf) [reply L/libgraphicsmagic/build_tarballs.jl]: does this work now?

File: L/libgraphicsmagic/build_tarballs.jl [hunk 3]
```diff
@@ -34,8 +34,19 @@ products = [
...
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
+    Dependency(PackageSpec(name="libwebp_jll", uuid="c5f90fcd-3b7e-5836-afba-fc50a0988cb2"))
+    Dependency(PackageSpec(name="libpng_jll", version=v"1.6.38", uuid="b53b4c65-9356-5827-b1ea-8c7a1a84506f"))
+    Dependency(PackageSpec(name="FreeType2_jll", uuid="d7e528f0-a631-5988-bf34-fe36492bcfd7"))
+    Dependency(PackageSpec(name="Ghostscript_jll", uuid="61579ee1-b43e-5ca0-a5da-69d92c66a64b"))
+    Dependency(PackageSpec(name="Xorg_libXext_jll", uuid="1082639a-0dae-5f34-9b06-72781eeb8cb3"))
+    Dependency(PackageSpec(name="Xorg_libSM_jll", uuid="c834827a-8449-5923-a945-d239c165b7dd"))
+    Dependency(PackageSpec(name="Libtiff_jll", uuid="89763e89-9b03-5906-acba-b20f662cd828"))
+    Dependency(PackageSpec(name="LittleCMS_jll", uuid="d3a379c0-f9a3-5b72-a4c0-6bf4d2e8af0f"))
+    Dependency(PackageSpec(name="Bzip2_jll", version=v"1.0.7", uuid="6e34b625-4abd-537c-b88f-471c36dfa7a0"))
```
Comment (giordano) [inline L/libgraphicsmagic/build_tarballs.jl:46]: This is not what was written there:
```suggestion
    Dependency("Bzip2_jll", v"1.0.7"; compat="1.0.7")
```

Comment (Moelf) [reply L/libgraphicsmagic/build_tarballs.jl]: sorry, the wizards spit out the full version so I tend to respect that format.

File: L/libgraphicsmagic/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply L/libgraphicsmagic/build_tarballs.jl]: No, it'll work after I'll merge https://github.com/JuliaRegistries/General/pull/37269 (which will happen after I'll fix all the compat bounds everywhere)

File: L/libgraphicsmagic/build_tarballs.jl (hunk 3 above)
Comment (giordano) [reply L/libgraphicsmagic/build_tarballs.jl]: The `PackageSpec` was fine, but not how you specified the version

File: L/libgraphicsmagic/build_tarballs.jl (hunk 1 above)
Comment (Moelf) [reply L/libgraphicsmagic/build_tarballs.jl]: ok, no rush, just merge whenever the rest of the landscape is ready. This is only used by my `timg` anyways, and I'm not in a rush.

Comment (giordano) [reply L/libgraphicsmagic/build_tarballs.jl]: For the record, with #3054 I managed to not break the ABI of the libraries, so specifying the latest version isn't necessary, 1.37 is fine (if you want to specify any version at all, not needed)

**Pull request merged**